### ISSUE TITLE
Pull Request Comments are not Replaced on Subsequent Runs

### ISSUE THREAD

**Issue by canhorn** [pulumi/actions#633]
### What happened?

Pull Request comments are not replaced on subsequent runs.

When Summary/Details were implemented, they were added into the heading. https://github.com/pulumi/actions/blob/0c9ec5ff167fe048d2f54775bf7e19a559d6e133/src/libs/pr.ts#L18-L21

This probably causes the `comments.find` to never find the comment, and always create a new comment on the PR.

### Steps to reproduce

Trigger multiple Workflow run, with Comment on PR set to `true`.

```
jobs:
  preview_production:
    name: Preview Production Environment
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2

      - uses: actions/setup-dotnet@v1
        with:
          dotnet-version: 6.0.x

      - uses: pulumi/actions@v3
        with:
          command: preview
          stack-name: production
          comment-on-pr: true
          diff: true
```

### Expected Behavior

The first comment about a run to be replaced on subsequent runs.

### Actual Behavior

Multiple Comments added to PR, one for every run.

### Versions used

Current

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by shreecodes** [comment]
This issue has made some of our infrastructure PRs very noisy.

I noticed that #635 was merged. Are you folks planning a new release soon? 😄

**Comment by RobbieMcKinstry** [comment]
Hi @shreecodes! We're having an intern discussion about release cadence for this repo. I'm new to the team, so I need to come up to speed on the release process for this repo, but I hope we can cut a release soon since this seems to be an annoyance for a couple of people.

I'm going to close this issue, as it was fixed with #635, but I'll continue the conversation around cadence to #669.
